lib/Driver/Driver.cpp

```diff
@@ -1751,7 +1756,10 @@ static StringRef getOutputFilename(Compilation &C,
   // FIXME: Treat GeneratePCHJobAction as non-top-level (to get tempfile and not
   // use the -o arg) even though, based on ownership considerations within the
   // driver, it is stored as a "top level" JobAction.
-  if (isa<GeneratePCHJobAction>(JA)) {
+  if (auto *PCHAct = dyn_cast<GeneratePCHJobAction>(JA)) {
+    // For a persistent PCH use the PCH output directory as the -o arg.
+    if (PCHAct->isPersistentPCH())
+      return PCHAct->getPersistentPCHDir();
```

Comment:
To determine the full filename for the persistent PCH we use hashes of the swift flags and the clang instance hash. Unfortunately that means that the driver cannot determine the filename in advance since it requires creating a clang compiler instance.
So the driver, for persistent PCH, passes `-emit-pch -pch-output-dir <path>` and the frontend invocation is figuring out the filename to write the PCH as.

But it doesn't seem necessary to have the output-dir as 'output' since the job is keeping track of it anyway, I'll look into change this part.